tools/gulp/packaging/build-config.ts

```diff
@@ -0,0 +1,17 @@
+/** Interface for the build config. */
+export interface BuildConfig {
+  projectVersion: string;
+  projectDir: string;
+  packagesDir: string;
+  outputDir: string;
+  licenseBanner: string;
+}
+
+/** Build config being used to build package output. */
+export let buildConfig: BuildConfig;
+
+/** Configures the build by setting the build config. */
+export function configureBuild(config: BuildConfig) {
```

Comment:
In those cases it makes the most sense to me to pass the data around. Ultimately this is an acknowledgement that the build config _is_ data, and not a set of code constants.